com.civicrm.hrjobroles/tests/phpunit/api/v3/ContactHRJobRoleTest.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+use CRM_HRCore_Test_Fabricator_Contact as ContactFabricator;
+use CRM_Hrjobcontract_Test_Fabricator_HRJobContract as HRJobContractFabricator;
+use CRM_Hrjobroles_BAO_HrJobRoles as HrJobRoles;
+use CRM_Hrjobroles_Test_Fabricator_HrJobRoles as HRJobRolesFabricator;
+
+/**
+ * Class api_v3_ContactHRJobRoleTest
+ *
+ * @group headless
+ */
+class api_v3_ContactHRJobRoleTest extends PHPUnit_Framework_TestCase implements HeadlessInterface, TransactionalInterface {
+
+  private $entity = 'ContactHrJobRoles';
+  private $action = 'get';
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+                     ->install('uk.co.compucorp.civicrm.hrcore')
+                     ->install('org.civicrm.hrjobcontract')
+                     ->installMe(__DIR__)
+                     ->apply();
+  }
+
+  public function testAccessAjaxPermissionIsRequiredToAccessTheGetAction() {
+    $contactID = 1;
+    $this->registerCurrentLoggedInContactInSession($contactID);
+
+    CRM_Core_Config::singleton()->userPermissionClass->permissions = [];
+
+    $this->setExpectedApiPermissionException('access AJAX API');
+    civicrm_api3($this->entity, $this->action, ['check_permissions' => true]);
+  }
+
+  public function testTheGetActionReturnsOnlyTheAllowedFields() {
```

Comment:
What about a test that adds the fields to return to the API and some of the passed fields should be the fields that are not supposed to be returned.